File: S1-voluntario/Programa.rb
require_relative "Cliente.rb"

module S1
	public
	class Programa

		def self.main
			
			cliente = Cliente.new

			
			puts "¿Cuantas bicicletas conforman la carrera?: "
			n = gets.to_i
			puts "\n"
			
			carreraCarretera = cliente.crearCarreraCarretera(n)
			carreraMontana = cliente.crearCarreraMontana(n)

			puts "Comienza la carrera de " + carreraCarretera.getTipo
			carreraCarretera.comenzar()
			puts
			puts "La carrera de " + carreraCarretera.getTipo + " ha finalizado"
			puts

			puts "Comienza la carrera de " + carreraMontana.getTipo
			carreraMontana.comenzar()
			puts
			puts "La carrera de " + carreraMontana.getTipo + " ha finalizado"
		end
	end
	Programa.main
end

File: P1/S1/Programa.java
package S1;

import java.util.Scanner;

public class Programa {
    public static void main(String[] args) {
        int n = 0;
        Scanner in = new Scanner(System.in);

        Cliente cliente = new Cliente();

        Carrera carreraMontana;
        Carrera carreraCarretera;

        while(n != -1)
        {
            System.out.println("\n¿Cuantas bicicletas conforman la carrera? (-1 para finalizar): ");
            n = in.nextInt(); in.skip("\n");

            carreraCarretera = cliente.crearCarreraCarretera(n);
            carreraMontana = cliente.crearCarreraMontana(n);

            carreraCarretera.comenzar();
            carreraMontana.comenzar();

            carreraCarretera.finalizar();
            carreraMontana.finalizar();
        }
        in.close();
    }
}